nucypher/cli/characters/alice.py

```diff
@@ -7,44 +7,170 @@
+
+        if federated_only and geth:
+            raise click.BadOptionUsage(
+                option_name="--geth",
+                message="--federated-only cannot be used with the --geth flag")
+
+        # Managed Ethereum Client
+        eth_node = NO_BLOCKCHAIN_CONNECTION
+        if geth:
+            eth_node = actions.get_provider_process()
+            provider_uri = ETH_NODE.provider_uri(scheme='file')
+
+        self.dev = dev
+        self.domains = {network} if network else None
+        self.provider_uri = provider_uri
+        self.geth = geth
+        self.federated_only = federated_only
+        self.eth_node = eth_node
+        self.pay_with = pay_with
+        self.discovery_port = discovery_port
+        self.registry_filepath = registry_filepath
+        self.middleware = middleware
+
+    def create_config(self, emitter, config_file):
+
+        if self.dev:
+            return AliceConfiguration(
+                emitter=emitter,
+                dev_mode=True,
+                network_middleware=self.middleware,
+                domains={TEMPORARY_DOMAIN},
+                provider_process=self.eth_node,
+                provider_uri=self.provider_uri,
+                federated_only=True) # FIXME: add error if --federated-only is not True?
 
+        else:
+            try:
+                return AliceConfiguration.from_configuration_file(
+                    emitter=emitter,
+                    dev_mode=False,
+                    network_middleware=self.middleware,
+                    domains=self.domains,
+                    provider_process=self.eth_node,
+                    provider_uri=self.provider_uri,
+                    filepath=config_file,
+                    rest_port=self.discovery_port,
+                    checksum_address=self.pay_with,
+                    registry_filepath=self.registry_filepath)
+            except FileNotFoundError:
+                return actions.handle_missing_configuration_file(
+                    character_config_class=AliceConfiguration,
+                    config_file=config_file)
+
+    def generate_config(self, emitter, config_root, poa, light, m, n, duration_periods, rate):
+
+        if self.dev:
+            raise click.BadArgumentUsage("Cannot create a persistent development character")
+
+        if not self.provider_uri and not self.federated_only:
+            raise click.BadOptionUsage(
+                option_name='--provider',
+                message="--provider is required to create a new decentralized alice.")
+
+        pay_with = self.pay_with
+        if not pay_with and not self.federated_only:
+            pay_with = select_client_account(
+                emitter=emitter,
+                provider_uri=self.provider_uri)
+
+        return AliceConfiguration.generate(
+            password=get_nucypher_password(confirm=True),
+            config_root=config_root, # FIXME: or DEFAULT_CONFIG_ROOT?
```

Comment:
Fixed in the new rebase. `or` is not needed here, since `config_root` is never `None` (it is set to the value from the config object in the calling function).